### PR TITLE
Add support for unstable Python versions

### PR DESCRIPTION
**Pull request by MaksimZhukov**
In scope of this PR we've added support for unstable Python versions. We've reworked
- `python-builder` scripts
- `setup-template` scripts
- `python-tests`

Also we've added tests to cover `python-manifest-config.json` and updated git submodule

### CODE REVIEW

File: .github/workflows/manifest-config-validation.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Install Pester
+      shell: pwsh
+      run: |
+        Install-Module Pester -Force -Scope CurrentUser -RequiredVersion 4.10.1
```
Comment (al-cheb) [inline .github/workflows/manifest-config-validation.yml:21]: Does it make sense to remove version at all?

File: builders/build-python.ps1 [hunk 2]
```diff
@@ -49,13 +50,13 @@ function Get-PythonBuilder {
...
-    param (
-        [version] $Version,
+    param(
+        [semver] $Version,
         [string] $Architecture,
         [string] $Platform
     )
 
-    $Platform = $Platform.ToLower()  
+    $Platform = $Platform.ToLower()
```
Comment (al-cheb) [inline builders/build-python.ps1:59]: By default, all comparison operators are case-insensitive. You don't need this statement.

File: builders/python-version.psm1 [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    param(
+        [Parameter(Mandatory)]
+        [string] $label
+    )
+
+    switch ($label) {
+        "alpha" { return "a" }
+        "beta" { return "b" }
+        "rc" { return "rc" }
+        Default { throw "Invalid version label '$label'" }
```
Comment (al-cheb) [inline builders/python-version.psm1:16]: Use the lower case for default

File: builders/python-version.psm1 [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    param(
+        [Parameter(Mandatory)]
+        [semver] $version,
+        [char] $delimiter = "."
+    )
+
+    $majorVersion = $version.Major
+    $minorVersion = $version.Minor
+    $patchVersion = $version.Patch
```
Comment (al-cheb) [inline builders/python-version.psm1:34]: Why do we need these three variables?

`$nativeVersion = "{0}.{1}.{2}" -f $version.Major, $version.Minor, $version.Patch`

File: builders/python-version.psm1 [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    $nativeVersion = "${majorVersion}.${minorVersion}.${patchVersion}"
+
+    if ($version.PreReleaseLabel)
+    {
+        $preReleaseLabel = ($version.PreReleaseLabel).Split($delimiter)
```
Comment (al-cheb) [inline builders/python-version.psm1:40]: () - redundant

File: tests/python-tests.ps1 [hunk 6]
```diff
@@ -1,5 +1,5 @@
 param (
-    [Version] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+    [String] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
```
Comment (al-cheb) [inline tests/python-tests.ps1:2]: In the above script all types are in lower case.

Comment (MaksimZhukov) [reply tests/python-tests.ps1]: Thanks!

File: builders/python-version.psm1 (hunk 5 above)
Comment (MaksimZhukov) [reply builders/python-version.psm1]: Thanks!

File: builders/python-version.psm1 (hunk 4 above)
Comment (MaksimZhukov) [reply builders/python-version.psm1]: Thanks!

File: builders/python-version.psm1 (hunk 3 above)
Comment (MaksimZhukov) [reply builders/python-version.psm1]: Thanks!

File: builders/build-python.ps1 (hunk 2 above)
Comment (MaksimZhukov) [reply builders/build-python.ps1]: Thanks!

File: .github/workflows/manifest-config-validation.yml (hunk 1 above)
Comment (MaksimZhukov) [reply .github/workflows/manifest-config-validation.yml]: Thanks!

File: builders/build-python.ps1 [hunk 7]
```diff
@@ -49,13 +50,13 @@ function Get-PythonBuilder {
...
-    param (
-        [version] $Version,
+    param(
+        [semver] $Version,
         [string] $Architecture,
         [string] $Platform
     )
 
-    $Platform = $Platform.ToLower()  
+    $Platform = $Platform
```
Comment (maxim-lobanov) [inline builders/build-python.ps1:59]: Why we need it?

Comment (MaksimZhukov) [reply builders/build-python.ps1]: Thanks!

**Pull request merged**